hasher-matcher-actioner/hmalib/lambdas/fetcher.py

```diff
@@ -102,7 +98,7 @@ def lambda_handler(event, context):
             privacy_group,
             api.app_id,
             te_data_bucket,
-            config.s3_key_prefix,
+            config.s3_te_data_folder,
```

Comment:
YMMV: I nearly did this in the original diff, but I learned that s3 actually doesn't support folders, and any `ls` option actually lists you the entire set. The amazon UI shows you folders, but only because it has a convention on... key prefixes.

Reply:
I personally found most of the code easier to read when I treated it as a folder. Don't feel particularly strongly but I think it makes the code more understandable everywhere except the prefix search. AWS also clearly endorses thinking of s3 as a file system so I think it's alright for us to treat it as one in code.

Could add a comment to the one location where we use the folder name as a key prefix

Reply:
I agree that the concept of folders makes this easier to describe. What I'm worried about is that the concept doesn't describe the underlying system.

Here's amazon's documentation: https://docs.aws.amazon.com/AmazonS3/latest/userguide/organizing-objects.html

In the subpage on https://docs.aws.amazon.com/AmazonS3/latest/userguide/using-prefixes.html, we can see that the API doesn't use folder, and instead uses prefix. In fact, no API (https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html) mentions folder that I can find.

If you absolutely want to follow through with this change, please do it as a separate PR and grep and replace every usage of "prefix". I'm okay with folder if we're internally consistent.

Reply:
So I supposed the question is wether we expect partners to interact with the AWS UI or CLI/API more. I would personally expect UI or else API through our code which I will comment on.

Code grep shows no other uses of `prefix` in `hmalib`